File: evidence/generated-systems/golang/governance-on-golang/governance-on-golang/internal/model/Obligation.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Obligation Declaration
//==============================================================
type Obligation struct {
    gorm.Model
     ReferenceNumber                                    string
    DescriptionText                                    string
    RegulationId         *uint
    Regulation           *Regulation `gorm:"foreignKey:RegulationId"`
     Controls           []Control `gorm:"foreignKey:ControlsFromObligationId"`
     Policies           []Policy `gorm:"foreignKey:PoliciesFromObligationId"`
     Contracts           []Contract `gorm:"foreignKey:ContractsFromObligationId"`
    ObligationType                      ObligationType
    ReviewFrequency                      ControlFrequency

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/governance-on-rails/governanceonrails/app/models/obligation.rb
class Obligation < ApplicationRecord
  enum ObligationType: [:Regulatory, :Contractual, :PolicyDerived, :IndustryStandard]
  enum ReviewFrequency: [:Continuous, :Daily, :Weekly, :Monthly, :Quarterly, :Annually, :AdHoc]


  has_many :Regulation, class_name: 'Regulation'
  has_many :Controls, class_name: 'Control'
  has_many :Policies, class_name: 'Policy'
  has_many :Contracts, class_name: 'Contract'

end
